packages/reports/addon/models/bard-request/request.js

```diff
@@ -459,6 +459,27 @@ export default Fragment.extend(Validations, {
     return this.removeSort(sortObj);
   },
 
+  /**
+   * Removes the sort from the sort array using the metric model and parameters
+   *
+   * @method removeSortMetricByModelAndParam
+   * @param {DS.Model} metricModel
+   * @param {Object} [parameters] - parameters [optional]
+   * @returns {Void}
+   */
+  removeSortMetricByModelAndParam(metricModel, parameters) {
+    if (parameters) {
+      let canonicalizedMetric = canonicalizeMetric({
```

Comment:
This metric should have a computed property of 'canonicalName', see if you can use that instead